### ISSUE TITLE
Promise.all returns a null value when using BigInt in a Where condition in findUnique

### ISSUE THREAD

**Issue by sorilove** [prisma/prisma#18171]
### Bug description

When using findUnique, if you use a BigInt in the where condition, Promise.all returns a null value. If you use a data type other than BigInt, it returns a normal lookup result, not a null value. Even if you use a BigInt, it returns a normal lookup result if you don't use Promise.all.

findFirst is handled correctly with Promise.all, even when using a BigInt in the where condition.

Currently, this issue occurs when findUnique uses a BigInt in the where condition when using Promise.all.

This was not an issue in Prisma version 4.9.x, but has been an issue since 4.10.x.

### Prisma information

```prisma
// Add your schema.prisma

generator client {
  provider        = "prisma-client-js"
  output          = "../../clients/account"
  previewFeatures = []
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model ptest1 {
  id   BigInt @id @default(0)
  desc String @db.VarChar(50)
}

model ptest2 {
  id   Int @id
  desc String @db.VarChar(50)
}

```

```ts
// Add your code using Prisma Client
import { PrismaClient } from "../prisma/clients/account";

const prisma = new PrismaClient();

(async () => {

    const jobs1 = [];
    for (let i = 0; i < 3; ++i) {
        jobs1.push(
            new Promise<string | undefined>(async resolve => {
                const player = await prisma.ptest1.findUnique({
                    where: {
                        id: 1n
                    }
                });
                resolve(player?.desc);
            })
        );
    }

    const jobs2 = [];
    for (let i = 0; i < 3; ++i) {
        jobs2.push(
            new Promise<string | undefined>(async resolve => {
                const player = await prisma.ptest1.findFirst({
                    where: {
                        id: 1n
                    }
                });
                resolve(player?.desc);
            })
        );
    }

    const jobs3 = [];
    for (let i = 0; i < 3; ++i) {
        jobs3.push(
            new Promise<string | undefined>(async resolve => {
                const player = await prisma.ptest2.findUnique({
                    where: {
                        id: 1
                    }
                });
                resolve(player?.desc);
            })
        );
    }

    const result1 = await Promise.all(jobs1);  // null
    const result2 = await Promise.all(jobs2);  // not null
    const result3 = await Promise.all(jobs3);  // not null

})();

```

### Environment & setup

- OS: Windows, RockyOS
- Database: MySQL
- Node.js version: 18.14.2

### Prisma Version

```
prisma                  : 4.11.0
@prisma/client          : 4.11.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine 8fde8fef4033376662cad983758335009d522acb (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 8fde8fef4033376662cad983758335009d522acb (at node_modules\@prisma\engines\migration-engine-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.11.0-57.8fde8fef4033376662cad983758335009d522acb
Default Engines Hash    : 8fde8fef4033376662cad983758335009d522acb
Studio                  : 0.483.0
```

**Comment by janpio** [comment]
Probably a duplicate of https://github.com/prisma/prisma/issues/18096?

**Comment by sorilove** [comment]
> Probably a duplicate of #18096. Can you confirm?

Yes, you're right, it's the same issue.

**Comment by Weakky** [comment]
Hey, thanks for the report! This issue should be fixed by https://github.com/prisma/prisma-engines/pull/3818. I'm not entirely sure the fix will make it to the 4.12.0 release. So either this release or the next one.
